propolis/src/common.rs

```diff
@@ -44,12 +44,24 @@ enum ROInner<'a> {
     Buf(&'a mut [u8]),
     Ptr(*mut u8, usize),
 }
+
+/// Represents an abstract requested read operation.
+///
+/// Exposes an API with various "write" methods, which fulfill the request.
 pub struct ReadOp<'a> {
     inner: ROInner<'a>,
     offset: usize,
     write_offset: usize,
 }
 impl<'a> ReadOp<'a> {
+    /// Initializes a new read operation from a buffer.
+    ///
+    /// # Arguments
+    ///
+    /// - `op_offset`: An auxiliary offset stored within the operation.
+    /// - `buf`: A buffer which represents the "sink" of the read operation.
+    // TODO: Is there a better explanation for `op_offset`? It doesn't seem to
+    // be used in any calculations within this operation.
```

Comment:
op_offset is used to specify the offset of the operation within a given region.  It's not used in the calculations so much as being "along for the ride" as metadata that the emulation can use, rather than needing another function argument to pass along the offset.